SwiftPackage/Sources/AssessmentModel/ResultData/AssessmentResult.swift

```diff
@@ -0,0 +1,284 @@
+//
+//  AssessmentResult.swift
+//  
+//
+//  Copyright © 2017-2022 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+import JsonModel
+
+/// An ``AssessmentResult`` is the top-level ``ResultData`` for an assessment.
+public protocol AssessmentResult : BranchNodeResult {
+
+    /// A unique identifier for this run of the assessment. This property is defined as readwrite
+    /// to allow the controller for the task to set this on the ``AssessmentResult`` children
+    /// included in this run.
+    var taskRunUUID: UUID { get set }
+
+    /// The ``versionString`` may be a semantic version, timestamp, or sequential revision integer.
+    var versionString: String? { get }
+    
+    /// An identifier for the assessment model associated with this result. If included, this
+    /// is intended to match the identifier used by the services that requested running the
+    /// assessment. This could be a schedule identifier or an identifier in a different namespace
+    /// than the "task identifier" used by the assessment developers to identify their assessments.
+    var assessmentIdentifier: String? { get }
+    
+    /// An identifier that can be used either by the assessment developer or scientist to make to services.
```

Comment:
"...to make to services"? What does that mean?

Reply:
It means I'm including it for northwestern. ;)